python/cudf/CMakeLists.txt

```diff
@@ -24,72 +24,15 @@ project(
   LANGUAGES CXX CUDA
 )
 
-option(FIND_CUDF_CPP "Search for existing CUDF C++ installations before defaulting to local files"
-       OFF
-)
-option(USE_LIBARROW_FROM_PYARROW "Only use the libarrow contained in pyarrow" OFF)
-mark_as_advanced(USE_LIBARROW_FROM_PYARROW)
-
-# Find Python early so that later commands can use it
-find_package(Python 3.9 REQUIRED COMPONENTS Interpreter)
-
-# If the user requested it we attempt to find CUDF.
-if(FIND_CUDF_CPP)
-  include(rapids-cpm)
-  include(rapids-export)
-  include(rapids-find)
-  rapids_cpm_init()
+find_package(cudf "${RAPIDS_VERSION}" REQUIRED)
 
-  if(USE_LIBARROW_FROM_PYARROW)
-    # We need to find arrow before libcudf since libcudf requires it but doesn't bundle arrow
-    # libraries. These variables have no effect because we are always searching for arrow via
-    # pyarrow, but they must be set as they are required arguments to the function in
-    # get_arrow.cmake.
-    set(CUDF_USE_ARROW_STATIC OFF)
-    set(CUDF_ENABLE_ARROW_S3 OFF)
-    set(CUDF_ENABLE_ARROW_ORC OFF)
-    set(CUDF_ENABLE_ARROW_PYTHON OFF)
-    set(CUDF_ENABLE_ARROW_PARQUET OFF)
-    include(../../cpp/cmake/thirdparty/get_arrow.cmake)
-  endif()
-
-  find_package(cudf "${RAPIDS_VERSION}" REQUIRED)
-
-  # an installed version of libcudf doesn't provide the dlpack headers so we need to download dlpack
-  # for the interop.pyx
-  include(../../cpp/cmake/thirdparty/get_dlpack.cmake)
-else()
-  set(cudf_FOUND OFF)
-endif()
+# an installed version of libcudf doesn't provide the dlpack headers so we need to download dlpack
```

Comment:
Do we think this is a bug in libcudf packaging? Should libcudf ship dlpack headers (or express this dependency some other way with CMake)?

Reply:
I don't know. This was in the diff before I took this PR over and I haven't gotten to this yet (note this is still in draft). On my list to look into.

Reply:
Ok I looked into this.

Notice that this is in the diff only because it's being moved around... it's not new behavior introduced in this PR. Looks like it's been in `cudf` since November 2022: #12139

Looks like the discussion on that PR led to a feature request to avoid needing to do this in places that use `libcudf`: #12175

That request is still open.

Happy to attempt fixing it as part of this work, but I'd need some help. Would the approach roughly be having `libcudf` wheels vendor the dlpack headers via including them in the `cudf::cudf` export sets, similar to what `librmm` does with `nvtx3`?

```cmake
rapids_cpm_nvtx3(BUILD_EXPORT_SET rmm-exports INSTALL_EXPORT_SET rmm-exports)
```

([rmm code link](https://github.com/rapidsai/rmm/blob/66320edf53336bf7a97c542fe7d45a60eb7811b5/cmake/thirdparty/get_nvtx.cmake#L19))

Reply:
Let’s mark #12175 for future build team work but not block this PR on that.

Reply:
Ok sounds good, thanks. I've added that to the build team board (in "Backlog").

Reply:
Yeah this is #12175. Let's not try to address that here. The ideal solution is not obvious.